src/main/java/com/denizenscript/clientizen/events/server/ScreenOpenCloseServerEvent.java

```diff
@@ -0,0 +1,66 @@
+package com.denizenscript.clientizen.events.server;
+
+import com.denizenscript.clientizen.network.DataDeserializer;
+import com.denizenscript.clientizen.network.DataSerializer;
+import net.minecraft.client.gui.screen.GameMenuScreen;
+import net.minecraft.client.gui.screen.Screen;
+import net.minecraft.client.gui.screen.advancement.AdvancementsScreen;
+import net.minecraft.client.gui.screen.ingame.CreativeInventoryScreen;
+import net.minecraft.client.gui.screen.ingame.InventoryScreen;
+import net.minecraft.client.gui.screen.option.OptionsScreen;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+public class ScreenOpenCloseServerEvent extends ServerEvent {
+
+	public static ScreenOpenCloseServerEvent instance;
+
+	public static Map<String, Class<?>> TYPE_MAP = new HashMap<>();
+
+	static {
+		TYPE_MAP.put("inventory", InventoryScreen.class);
+		TYPE_MAP.put("creative", CreativeInventoryScreen.class);
+		TYPE_MAP.put("pause", GameMenuScreen.class);
+		TYPE_MAP.put("options", OptionsScreen.class);
+		TYPE_MAP.put("advancements", AdvancementsScreen.class);
+	}
+
+	public static List<String> listenToOpenScreens = new ArrayList<>();
+	public static List<String> listenToCloseScreens = new ArrayList<>();
+
+	public ScreenOpenCloseServerEvent() {
+		instance = this;
+		id = "PlayerOpenCloseScreen";
+	}
+
+	public void handleScreenChange(Screen screen, Screen otherScreen, boolean open) {
+		for (String type : open ? listenToOpenScreens : listenToCloseScreens) {
+			if (type.equals("inventory") && screen instanceof CreativeInventoryScreen) {
+				continue;
+			}
+			if (TYPE_MAP.get(type).isInstance(screen)) {
+				DataSerializer serializer = new DataSerializer()
```

Comment:
Unnecessary variable, can just initialize it inside `fire`

Reply:
Doesn't look as nice but I'll follow the existing convention